torch/_inductor/models/mm_kernel_prediction_model.py

```diff
@@ -0,0 +1,678 @@
+
+                # Validate model and update metrics
+                test_mse, test_log_mse, best_state_dict = validate(
+                    model=model,
+                    criterion=criterion,
+                    test_x=test_x,
+                    test_y=test_y,
+                    best_mse_seen=min(test_mses) if len(test_mses) > 0 else np.inf,
+                    best_state_dict=best_state_dict,
+                    epoch=epoch,
+                    verbose=verbose,
+                )
+                epochs.append(epoch)
+                test_mses.append(test_mse)
+                test_log_mses.append(test_log_mse)
+                train_mses.append(train_mse)
+                should_stop = test_mses[-1] > 3 * min(test_mses)
+
+                if should_stop:
+                    print(f"Test MSE of {test_mse} > 3 * {min(test_mses)}")
+                    break
+
+    except KeyboardInterrupt:
+        max_epoch = 0 if len(epochs) == 0 else epochs[-1]
+        print(f"KeyboardInterrupt after {max_epoch} epochs")
+    mse_traces = {
+        "test_mses": test_mses,
+        "test_log_mses": test_log_mses,
+        "train_mses": train_mses,
+    }
+    model.load_state_dict(state_dict=best_state_dict)
+    return model.to(device="cpu").eval(), epochs, mse_traces
+
+
+def get_total_gb_feature(df: pd.DataFrame) -> pd.Series:
+    """
+    Calculate the total gigabytes feature from the dataframe.
+
+    Args:
+        df: DataFrame containing the necessary columns for calculation
+
+    Returns:
+        Series containing the calculated total gigabytes
+    """
+    # Calculate memory access in bytes
+    m, n, k = df["dim_m"], df["dim_n"], df["dim_k"]
+    dtype_size = df["dtype_size"] / 8  # Convert bits to bytes
+
+    # A: m×k, B: k×n, C: m×n
+    return ((m * k + k * n + m * n) * dtype_size) / 1e9  # Convert to GB
+
+
+def get_total_gflop_feature(df: pd.DataFrame) -> pd.Series:
+    """
+    Calculate the total gigaflops feature from the dataframe.
+
+    Args:
+        df: DataFrame containing the necessary columns for calculation
+
+    Returns:
+        Series containing the calculated total gigaflops
+    """
+    # For matrix multiplication, flops = 2 * m * n * k
+    m, n, k = df["dim_m"], df["dim_n"], df["dim_k"]
+    return (2 * m * n * k) / 1e9  # Convert to GFLOP
+
+
+class ModelWrapper:
+    """
+    Wrapper for the neural network model that handles encoding inputs and decoding outputs.
+
+    This class provides methods to prepare inputs for the model and interpret its outputs,
+    handling the necessary standardization and feature engineering.
+    """
+
+    def __init__(self) -> None:
+        """Initialize the model wrapper with the pre-trained model and standardization parameters."""
+        start_time = time.time()
+        self.model = NeuralNetwork(
+            n_inputs=12, hidden_layer_widths=[2**8 for _ in range(6)]
+        )
+        self.model.load_state_dict(torch.load(MODEL_PATH))
```

Comment:
Aside: In a future update I hope to save the coefficients on the model so that `load_state_dict` is all that's needed, rather than hardcoding the numbers in L480-L509